Source/Core/Core/PatchEngine.cpp

```diff
@@ -214,6 +216,11 @@ void LoadPatches()
   Common::IniFile globalIni = sconfig.LoadDefaultGameIni();
   Common::IniFile localIni = sconfig.LoadLocalGameIni();
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+  std::lock_guard lg{AchievementManager::GetInstance().GetLock()};
+  AchievementManager::GetInstance().SetGameIniId(sconfig.GetGameID());
+#endif  // USE_RETRO_ACHIEVEMENTS
```

Comment:
I think you want to create a scope for the lock here?

Reply:
Haha I just mentally processed the ifdef block as a scope, oops.